### ISSUE TITLE
Support of conditional nfts for nested objects

### ISSUE THREAD

**Issue by vikiival** [kodadot/packages#19]
Support at least level one

https://blog.logrocket.com/how-to-use-keyof-operator-typescript/

```ts
export type SomeNFT = {
  currentOwner: string
  id: string
  issuer: string
  meta?: {
    description: string
    name: string
    image: string
  }
}
```

**Comment by vikiival** [comment]
Reading:

https://javascript.plainenglish.io/advanced-typescript-type-level-nested-object-paths-7f3d8901f29a

https://stackoverflow.com/questions/58434389/typescript-deep-keyof-of-a-nested-object

https://dev.to/pffigueiredo/typescript-utility-keyof-nested-object-2pa3

https://www.delftstack.com/howto/typescript/use-the-keyof-operator-in-typescript/

https://errorsfixing.com/typescript-deep-keyof-of-a-nested-object/

**Comment by vikiival** [comment]
Implemented between a602177d0072bd86332077d5cfffdccb46b8432b and 6d32f017ff61f2f60077bc5d9b452ab6c41e453a
